bindings/java/src/main/java/org/ucx/jucx/ucp/UcpParams.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (C) Mellanox Technologies Ltd. 2019. ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+package org.ucx.jucx.ucp;
+
+import static org.ucx.jucx.ucs.UcsBits.UCS_BIT;
+
+/**
+ * Tuning parameters for UCP library.
+ *
+ * The structure defines the parameters that are used for
+ * UCP library tuning during UCP library {@link UcpContext} "initialization".
+ *
+ * <p>UCP library implementation uses the {@link UcpParams.UcpFeature} "features"
+ * parameter to optimize the library functionality that minimize memory
+ * footprint. For example, if the application does not require send/receive
+ * semantics UCP library may avoid allocation of expensive resources associated with
+ * send/receive queues.
+ */
+public class UcpParams {
+
+    @Override
+    public String toString() {
+        return "UcpParams{"
+            + "fieldMask=" + fieldMask
+            + ", features=" + features
+            + ", tagSenderMask=" + tagSenderMask
+            + ", mtWorkersShared=" + mtWorkersShared
+            + ", estimatedNumEps=" + estimatedNumEps
+            + '}';
+    }
+
+    /**
+     * UCP context parameters field mask.
+     *
+     * <p>The enumeration allows specifying which fields in {@link UcpParams} are
+     * present. It is used for the enablement of backward compatibility support.
+     */
+    private enum UcpParamField {
+        UCP_PARAM_FIELD_FEATURES(UCS_BIT(0L)),
+        UCP_PARAM_FIELD_TAG_SENDER_MASK(UCS_BIT(4L)),
```

Comment:
can we take these constants from C somehow? why have to redefine in Java?

Reply:
This is typical practice in java<->C interop:
https://github.com/zrlio/disni/blob/d0a2ad6d20a1c931d0b7e7b7404c30f288eaca06/src/main/java/com/ibm/disni/verbs/IbvWC.java#L64

 It's possible to put a bunch of booleans in java and do all bitset logic in jni, but as for me this makes code more concise.

Reply:
can we add assertions in JNI that JAVA constant values are equal to the equivalent C constants?

Reply:
I thought it's backward compatible  - new values will have new mask. OK we can set these values from jni, the same as in ompi: https://github.com/open-mpi/ompi/blob/master/ompi/mpi/java/c/mpi_Constant.c#L60

Reply:
Now setting all constants from JNI.